src/Ocelot/Configuration/Validator/FileConfigurationFluentValidator.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FluentValidation;
+using Microsoft.AspNetCore.Authentication;
+using Ocelot.Configuration.File;
+using Ocelot.Errors;
+using Ocelot.Responses;
+
+namespace Ocelot.Configuration.Validator
+{
+    public class FileConfigurationFluentValidator : AbstractValidator<FileConfiguration>, IConfigurationValidator
+    {
+        public FileConfigurationFluentValidator(IAuthenticationSchemeProvider authenticationSchemeProvider)
+        {
+            RuleFor(configuration => configuration.ReRoutes)
+                .SetCollectionValidator(new ReRouteFluentValidator(authenticationSchemeProvider));
+            RuleForEach(configuration => configuration.ReRoutes)
+                .Must((config, reRoute) => !IsDuplicateIn(reRoute, config.ReRoutes))
+                .WithMessage((config, reRoute) => $"duplicate downstreampath {reRoute.UpstreamPathTemplate}");
+        }
+
+        private static bool IsDuplicateIn(FileReRoute reRoute, List<FileReRoute> routes)
+        {
+            var reRoutesWithUpstreamPathTemplate = routes.Where(r => r.UpstreamPathTemplate == reRoute.UpstreamPathTemplate).ToList();
+            var hasEmptyListToAllowAllHttpVerbs = reRoutesWithUpstreamPathTemplate.Any(x => x.UpstreamHttpMethod.Count == 0);
+            var hasDuplicateEmptyListToAllowAllHttpVerbs = reRoutesWithUpstreamPathTemplate.Count(x => x.UpstreamHttpMethod.Count == 0) > 1;
+
+            var hasSpecificHttpVerbs = reRoutesWithUpstreamPathTemplate.Any(x => x.UpstreamHttpMethod.Count != 0);
+            var hasDuplicateSpecificHttpVerbs = reRoutesWithUpstreamPathTemplate.SelectMany(x => x.UpstreamHttpMethod).GroupBy(x => x.ToLower()).SelectMany(x => x.Skip(1)).Any();
+            if (hasDuplicateEmptyListToAllowAllHttpVerbs || hasDuplicateSpecificHttpVerbs || (hasEmptyListToAllowAllHttpVerbs && hasSpecificHttpVerbs))
+            {
```

Comment:
This used to return a DownstreamPathTemplateAlreadyUsedError we should probably keep this and add more errors for the things we are testing so the user gets a good exception message.

Reply:
i think it's not necessary for startup validation check return a specific type error. any way, it will not start while validation is not success and what most important is we give the user where the error occur and how to fix it. not as run time check, i think is ok to return a error message and point out the error.

Reply:
But how will the user know what the problem is? Don't we need to say something like..

This was the duplicate ReRoute xxxx
This ReRoute had a scheme in its path template

etc

Reply:
yes, this is exactly the WithMessage method do.

Reply:
all error message is populate here. https://github.com/Eilyyyy/Ocelot/blob/5512f2f247c205d5afc7350bd6271a91daa18234/src/Ocelot/Configuration/Validator/FileConfigurationFluentValidator.cs#L30

Reply:
OK cool, makes sense